eventsourced/src/lib.rs

```diff
@@ -352,25 +377,32 @@ where
     Self: Debug,
     E: EventSourced,
 {
-    fn handle_cmd(&self, id: &E::Id, state: &E) -> Result<E::Evt, BoxedAny>;
-
-    fn make_reply(&self, id: &E::Id, state: &E, evt: E::Evt) -> BoxedAny;
+    #[allow(clippy::type_complexity)] // internal type, who cares?
+    fn handle_cmd(
+        &self,
+        id: &E::Id,
+        state: &E,
+    ) -> Result<(E::Evt, Box<dyn FnOnce(&E) -> BoxedAny + Send + Sync>), BoxedAny>;
 }
 
 impl<C, E, Reply, Error> ErasedCmd<E> for C
 where
     C: Cmd<E, Reply = Reply, Error = Error>,
-    E: EventSourced,
+    E: EventSourced + 'static,
     Reply: Send + 'static,
     Error: Send + 'static,
 {
-    fn handle_cmd(&self, id: &E::Id, state: &E) -> Result<E::Evt, BoxedAny> {
-        let result = self.handle_cmd(id, state);
-        result.map_err(|error| Box::new(error) as BoxedAny)
-    }
-
-    fn make_reply(&self, id: &E::Id, state: &E, evt: E::Evt) -> BoxedAny {
-        Box::new(self.make_reply(id, state, evt))
+    fn handle_cmd(
+        &self,
+        id: &E::Id,
+        state: &E,
+    ) -> Result<(E::Evt, Box<dyn FnOnce(&E) -> BoxedAny + Send + Sync>), BoxedAny> {
```

Comment:
I have not checked this yet, but I assume one could capture the command or parts of it or the to be emitted event (clone if needed)?

Reply:
That's the whole idea, exactly. I try converting `handle_cmd` to owning `self`, then it will be even more convenient to just consume the command finally in the `make_reply` closure.

Reply:
Nice.